src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkConfig.java

```diff
@@ -1,22 +1,75 @@
 package com.azure.cosmos.kafka.connect.sink;
 
+import com.azure.cosmos.kafka.connect.sink.id.strategy.AbstractIdStrategyConfig;
+import com.azure.cosmos.kafka.connect.sink.id.strategy.IdStrategy;
+import com.azure.cosmos.kafka.connect.sink.id.strategy.ProvidedInValueStrategy;
 import org.apache.kafka.common.config.ConfigDef;
 
 import java.util.Map;
 
 import com.azure.cosmos.kafka.connect.CosmosDBConfig;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Width;
+import org.apache.kafka.common.config.ConfigException;
 
 /**
  * Contains settings for the Kafka ComsosDB Sink Connector
  */
 
 @SuppressWarnings ({"squid:S1854", "squid:S2160"})  // suppress unneeded int *groupOrder variables, equals method
 public class CosmosDBSinkConfig extends CosmosDBConfig {
+    public static final String ID_STRATEGY_CONFIG = AbstractIdStrategyConfig.ID_STRATEGY;
+    public static final Class<?> ID_STRATEGY_DEFAULT = ProvidedInValueStrategy.class;
+    public static final String ID_STRATEGY_DOC = "";
+    public static final String TEMPLATE_CONFIG_DISPLAY = "ID Strategy";
+
+    private IdStrategy idStrategy;
+
     public CosmosDBSinkConfig(ConfigDef config, Map<String, String> parsedConfig) {
         super(config, parsedConfig);
+
+        this.idStrategy = createIdStrategy();
     }
 
     public CosmosDBSinkConfig(Map<String, String> parsedConfig) {
-        super(getConfig(), parsedConfig);
+        this(getConfig(), parsedConfig);
+    }
+
+    public static ConfigDef getConfig() {
+        ConfigDef result = CosmosDBConfig.getConfig();
+
+        final String groupName = "ID Strategy Parameters";
+        int groupOrder = 0;
+
+        result
+            .define(
+                ID_STRATEGY_CONFIG,
```

Comment:
@NathanNam this config allows the user to specify any class implementing the `IdStrategy` interface to use for generating the ID. The user will have to specify the full name of the class so it can be instantiated through reflection. Something like `com.azure.cosmos.kafka.connect.sink.id.strategy.ProvidedInValueStrategy.class` . This semantic is similar to how we specify things like data format and partitioner on S3 sink connector. It also allows for users to implement their own `IdStrategy` and drop it in the plugin path for the connector to use.

Reply:
+1